src/sid/contacts.py

```diff
@@ -342,27 +362,32 @@ def _calculate_infections_by_recurrent_contacts(
                     # never happens that i is infectious but not immune
                     if not immune[j] and recurrent_contacts[j, cm] > 0:
                         # j is infected depending on its own susceptibility.
+                        virus_strain_i = virus_strain[i]
+                        virus_multiplier_i = virus_strains_multipliers[virus_strain_i]
                         multiplier = infection_probability_multiplier[j]
-                        is_infection = boolean_choice(prob * multiplier)
+                        is_infection = boolean_choice(
+                            prob * multiplier * virus_multiplier_i
```

Comment:
We need better names here. I suggest:
prob -> base_probability
multiplier -> susceptibility_factor / susceptibility_multiplier
virus_multiplier -> contagiousness_factor / contagiousness_multiplier

Then pull the product into a separate line:
`individual_infection_risk = base_probability * susceptibility_factor * contagiousness_factor`

I would personally prefer the suffix `_factor` because its shorter and we already have so many multipliers in policies. But `_multiplier` is more aligned with the argument names you chose and thus probably better.